js/h5p.js

```diff
@@ -1400,6 +1406,182 @@ H5P.openEmbedDialog = function ($element, embedCode, resizeCode, size, instance)
   dialog.open();
 };
 
+/**
+ * Recursive internal function that compares two objects, utilizing deepEqual
+ * 
+ * 
+ * @param {object} a The first object to be compared
+ * @param {object} b The second object to be compared
+ * @param {object} aStack First object internal object stack
+ * @param {object} bStack Second object internal object stack
+ */
+H5P.eq = function(a, b, aStack, bStack) {
+
+  //identical objects are equal but not identical, ex. 0 === -0
+  if(a === b) {
+    return a !== 0 || 1 / a === 1 / b;
+  }
+
+  //null and undefined only equal to themselves
+  if(a === null || b === null) {
+    return false;
+  }
+
+  //NaN equivalent but not reflexive
+  if(a !== a) {
+    return b !== b;
+  }
+
+  //exhaust primitives
+  var type = typeof a;
+  if(type !== 'function' && type !== 'object' && typeof b !== 'object') {
+    return false;
+  }
+  return H5P.deepEqual(a, b, aStack, bStack);
+}
+
+
+/**
+ * Internal recursive function that determines if two objects are equivalent
+ * 
+ * Returns true if equal and false otherwise
+ * 
+ * @param {object} aObject First object being compared
+ * @param {object} bObject Second object being compared
+ * @param {object} aStack a stack reprentation of the first objects' internal object stack
+ * @param {object} bStack a stack reprentation of the second objects' internal object stack
+ */
+H5P.deepEqual = function (aObject, bObject, aStack, bStack) {
```

Comment:
if nothing calls deepEqual (including deepEqual itself), and both functions require the same arity, why have two functions?